### DISCUSSION TITLE
Cmake Problem

### DISCUSSION THREAD

**Discussion by Lukmin1999** [musescore/MuseScore#12083] [Q&A] [answered]
Hello. I'm not sure if this is worth putting into Issues, since it is likely a problem on my end.

I'm trying to build Musescore on my computer at the moment. I've downloaded qt version 5.15.2, edited my environment variables, and downloaded and installed Cmake. However, when I open Musescore file on Visual Studio, there don't appear to be any solutions. When I try to make them myself by running cmake -P build.cmake -DCMAKE_BUILD_TYPE=Release, I get
`could not find any instance of visual studio`

I've tried changing the cmake build text to look for instances of Visual Studio 2022, since that is  the version I have. I receive similar problems. What might I be doing wrong?

**Comment by Jojo-Schmitz** [comment, marked as answer]
Not sure we're ready yet for VS 2022, VS 2019 is what's currently being used

**Comment by HemantAntony** [comment]
This is a good place for your question (Our discord server would have also been good). Though I doubt I will be able to answer you question, can you tell me which line you changed and to what?

**Comment by Jojo-Schmitz** [comment]
msvc_build.bat is only prepared for VS 2017 and VS 2019 so far, and CMakeSettings.json and build.cmake only for VS 2019. At least the latter 2 would need to get adjusted, the former only if you're using that (I do for the initial setup)

**Comment by cbjeukendrup** [comment]
`cmake -P build.cmake -DCMAKE_BUILD_TYPE=Release -G "Visual Studio 17 2022"` might help.

**Reply by Jojo-Schmitz** [reply]
Any plans to officially move to VS 2022?

**Reply by cbjeukendrup** [reply]
I haven't heard of any, but maybe. I don't expect the change will be a big deal for us.

**Reply by Jojo-Schmitz** [reply]
Probably not, but at least 3 files in the repo need to get changed for that (see above), and 2 of those are making it neccessary for everyone to switch (including GitHub CI).
Might be a nice change to do now rather than after some release (Alpha, Beta, RC, final)?

Let's see, #12086

Apparently the changes from above mentioned PR doesn't require me to switch to VS2022 (which I haven't yet) nor needs VS 2022 on GitHub CI (it isn't there apparently. Edit: it is the windows-2022 environment)

Whether it'd work on a PC with VS 2022 remains to be tested ;-)

It works on mine. And on GitHub CI too now

**Reply by HemantAntony** [reply]
I'm trying to test it out now. I'll tell you

**Reply by HemantAntony** [reply]
It works for me too

**Reply by Lukmin1999** [reply]
Is this worth making an issue or feature request? If so, I'd be interested in taking up working on it. I'm not too familiar with the ins and outs of CMake, so I'm thinking this would be a good chance to do so while helping out.

**Reply by Lukmin1999** [reply]
By the way, I tried using 2019 VS, but now I get an error that says "could not find:sndfile"

**Reply by cbjeukendrup** [reply]
> "could not find:sndfile"

You'll also get this issue with VS 2022. For the solution, see this part of the MuseScore 3 compile instructions: https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-windows-visual-studio#1.7_Additional_dependencies
Unfortunately, that has not yet been added to the MuseScore 4 instructions (or has it, meanwhile?).

**Comment by Lukmin1999** [comment]
Based on the pull request I saw, should VS 2022 now be useable?

**Reply by cbjeukendrup** [reply]
The pull request has not been merged yet, but you can make the changes locally, so that you can at least continue compiling MuseScore!
